video/out/vo_gpu_next.c

```diff
@@ -1862,6 +1869,28 @@ static void update_lut(struct priv *p, struct user_lut *lut)
     talloc_free(lutdata.start);
 }
 
+static void update_hook_opts_dynamic(struct priv *p, const struct pl_hook *hook,
+                                     const struct mp_image *mpi)
+{
+    const struct { const char *name; double value; } opts[] = {
+        { "PTS", mpi->pts },
+    };
+
+    for (int i = 0; i < hook->num_parameters; i++) {
+        const struct pl_hook_par *hp = &hook->parameters[i];
+        for (int n = 0; n < MP_ARRAY_SIZE(opts); n++) {
+            if (strcmp(hp->name, opts[n].name) != 0)
+                continue;
+
+            switch (hp->type) {
+                case PL_VAR_FLOAT: hp->data->f = opts[n].value; break;
+                case PL_VAR_SINT:  hp->data->i = round(opts[n].value); break;
+                case PL_VAR_UINT:  hp->data->u = round(opts[n].value); break;
```

Comment:
It is UB to assign values form outside the variable range. Should be clampled here, also consider using `lrintf` that would help.